### ISSUE TITLE
VidCoder 5.1 Beta

### ISSUE THREAD

**Issue by spikaspika** [RandomEngy/VidCoder#473]
Version 5.1
The meaning of the program disappears. Users have passed the transition to the new interface, but the refusal of the “Advanced” tab has meant that the program has rolled up to the standard numerous utilities, of which there is a huge amount. Lost functionality and charm.
On the new version I feel like a loser with dull brains, who was forbidden to think and decide.
I remain on versions 3.16 and 4.29. It is a pity, the prospects were huge.

**Comment by RandomEngy** [comment]
You can still do everything you could do in the Advanced tab by entering settings in the "More options" field on the Video Encoding tab. The Advanced settings were just a fancy way of creating that options string.

The [x264 options documentation](https://en.wikibooks.org/wiki/MeGUI/x264_Settings) describes all the options you can edit there.

The "Advanced" tab was a bad way to do editing: it hasn't been updated in years, is missing dozens of settings, has bad defaults, doesn't do proper input validation, etc.

If you read the documentation and create your own options string you have more power than you did before.

**Comment by padi543** [comment]
Sorry,

I completely agree with spikaspika here:

Without a simple and fast way of setting encoder values, version 5.x is almost useless to me…

I understand the motivation behind the removal-decision, but just killing the advanced tab completely doesn't seem the right way:

- “_hasn't been updated in years_”
Yes, that's correct but also true for all other (free) encoder GUIs out there and for a very good reason: Most of these settings are “stable” and have proven to be crucial to tweak speed, quality and compatibility!

- “_missing dozens of settings_”
Agreed, again. Personally I have wished for some more or at least other settings in the advanced tab, but the existing text-field for additional commands was (and is) the perfect way to handle such specific custom changes.

- “_has bad defaults_”
Maybe true(?) but that's imho definitely not the purpose of a tab that's called “advanced” and is part of a freeware program!

- “_doesn't do proper input validation_”
And that's exactly the issue I have with a “generic”, primitive text-field/area for “More options”:
Batch-Encoding videos (at least on my machine, without any GPU acceleration) takes at least several hours, sometimes days… If I only find out the next morning, that most of the tasks failed because of a missing “;” or some other stupid typo, I have to question the actual advantage of using an encoder “GUI”.

- “_you have more power_”
Using the same argument a lot of other not-perfectly-working GUI options/features could be stripped away and replaced with more flexible and powerful command-line strings.


Sorry, that's just my opinion!


So here's my feature-request for v5.x:
- Some kind of support to additional encoding options, which does NOT require (but still allows) manual input ...
- Adding _standard_ options should not require entering any text or delimiters manually.
- Options should be selectable from an existing set of standard commands.
- A warning should be displayed for options not part of the Help/Documentation.
- Duplicate entries should be removed.
- At least for some (important/standard) options the values should be validated or simply selectable from a list of valid values.

**Comment by iamjaydear** [comment]
I agree - this change is a bit previous. One of the main things the Advanced tab provided for me was the ability to select various Quality, Encoder, Tune, Profile, etc. settings and see what changed in there. Then I could make my own little changes and see if I'd improved some aspect of the encode (or made it worse :) - it was a bit of a learning assistant if you like. Now it's gone and has dumbed down VidCoder as a result. If you are saying that the information in the Advanced tab was wrong, then it needs updating not eliminating. And I think simply referring us to a very geeky and complex wikibook is just unfriendly. If I could understand all the complexities of HandBrake I'd be an H.26x genius, but I'm not and I need the sort of help the Advanced tab provided.

**Comment by RandomEngy** [comment]
The "standard" options are Preset, Tune, Profile, Level. The vast majority of people should be able to accomplish anything they want from there. If you're delving into the individual x264 options you're basically signing up to try and craft settings that give you a better compression efficiency/encode time ratio than the developers of x264 were able to accomplish themselves. How many people can actually accomplish this? Not a lot. Not enough to make it worth it to expend a bunch of development effort trying to stuff every single setting into a UI. And to ask all the VidCoder translators to translate long explanation tooltips of everything.

> Using the same argument a lot of other not-perfectly-working GUI options/features could be stripped away and replaced with more flexible and powerful command-line strings.

Actually there are a lot of other features that already take command line strings: Every other video encoder, Detelecine, Yadif, Decomb, hqdn3d, NL-Means, UnSharp and LapSharp filters. Though nobody has ever asked me to create "advanced" GUIs for those. The model has been to supply some sane presets but let the user go wild with custom settings if they want. That way there isn't a ton of UI work needed every time an encoder adds or changes a parameter, 99.9% of users can accomplish their tasks easily but all the power is still there for people who want to delve deeper. I actually think VidCoder does a good job on the filter custom settings because it lists all the possible parameters in the tooltip and auto-populates the box with the "default" settings.

The options I do have UI for are there for a reason. You need to change the resolution based on your expected playback device. You need to switch on or off Anamorphic in case your player doesn't support it. You need to rotate your video if it came off your phone that way. You need to pick a different audio codec depending on player support, and different quality targets depending on your size requirements. UI has cost to implement, test and maintain and update and the benefit you get from it needs to be worth it. In my view what you get from tweaking x264 options is at best extremely marginal improvements over existing presets.

I have a full-time job, a family, a house and yard to maintain and other side software projects, so it really doesn't make sense to me to sink even more time into this and also commit to sinking a lot more and coming back to it on every x264 update.

@iamjaydear you can still see the final options string in a tooltip if you want to see what the presets/levels/profiles do. Also I'm trying to say that you _don't need to try and understand the Advanced settings_ . It's great that you like learning about the internals of x264, but VidCoder is meant to be a practical tool, not a learning assistant. If you want to sign up to do that, doing it via a GUI that someone else writes is not a good avenue:

- You won't see all the settings
- You'll be working on out-of-date information
- Also you're already in "geeky" territory by wanting to tweak this kind of stuff

Anyway, if this is really important to you, it could be brought back if:

- You do the work to bring it up to date
- You'll be around to keep it up to date
- It comes back in a different form. The huge field of boxes and sliders is not practical, especially if you're adding even more parameters in. If it ended up being some intellisense/autocomplete magic on the text field, where it suggests valid parameters and gives in-line hints about valid values that would be fine with me.

**Comment by RandomEngy** [comment]
![image](https://user-images.githubusercontent.com/9450861/51268358-cac01f00-1974-11e9-8a9f-98715b649a2c.png)

If you don't want to know the H264 internals I suggest sticking to the basic controls. Set the Level/Profile based on what your playback device can handle, choose a Tune based on the type of source material, and pick a preset based on how much encoding time you want to devote to increasing compression efficiency.

**Comment by padi543** [comment]
- Totally understand and agree that the existing „advanced-tab“ has no future:
Whoever tried to implement AND maintain something like this will understand what kind of unbelievable pita such a “nice & smart” GUI monster really is!!!
Anyone who disagrees: "_Go on, just program it! But also make sure that it's working after each and every Handbrake Update!!!_" … not only RandomEngy, we all would be very grateful!

- On the other hand:
The current layout of the “More settings” in v5.1 beta painfully, VERY painfully reminds of 20 years of cursing & shouting at the Windows dialog for maintaining the “Environment variables” …
https://www.ghacks.net/wp-content/uploads/2015/09/old-windows-environment-variables-editor.jpg


Is there really no alternative, no middle-way between both extremes?

... just one idea: 
hate to say it, but why not learn form Windows 10?
https://www.ghacks.net/wp-content/uploads/2015/09/windows-10-new-environment-variables-editor.jpg

(Source: [here](https://www.ghacks.net/2015/09/22/microsoft-improves-environment-variables-editor-in-latest-windows-10-build/))

**Comment by RandomEngy** [comment]
Souping up the property editor in some way like that sounds fine with me. I'd be happy to take that as a contribution to the project.

**Comment by spikaspika** [comment]
I understand and support the author of the program very well. Indeed, it is very difficult to combine many cases and at the same time develop the program, and even on a voluntary basis.
Therefore, not much upset. Indeed, in principle, the H264 and H265 processing algorithms are brought to near perfection, and if the Handbrake module does not drastically change, then you can simply update it by placing it in the program folder. And even if it changes, the quality of video processing that exists at the moment suits me perfectly. Even using the current versions of the program with the tab "advanced"? I am very pleased and I am very grateful to the author for his hard work.

**Comment by iamjaydear** [comment]
Thanks for the Full Parameter List screen grab above. I swear I was not seeing that popup! If I right-clicked all I saw was my added bits. Checking it again this afternoon I can see the whole list without a click! Maybe my PC needed to restart after the new v5.1 went in? Also, I was not aware that there are so many more presets now. I will be checking some of them out with a view to revamping my Picker settings. Thanks for VC and thanks for your support :)

**Comment by RandomEngy** [comment]
Copying and pasting x264 options strings is probably not a great idea in the first place. But the format is determined by HandBrake, so I can't change it.

**Comment by iamjaydear** [comment]
Yeah, if it won't support it there's no point. Before VidCoder I used WinFF (defunct now) and it was handy to be able to edit command strings. All the video and audio codes were in one long string. I guess HandBrake handles audio codes separately from video unlike FFmpeg.

**Comment by padi543** [comment]
**Related Feature Request:**
If It can be done easily a “_Validate_” feature could be helpful:
Just like the button “`Encode Preview Clip`” in the preview window 
the selected encoding options could be applied for a short test-encode … 

bad x264/x265 options should immediately result in an error 
(instead of hours later during the batch process)

**Comment by RandomEngy** [comment]
When you click "Encode Preview Clip" it will use the settings you've chosen.

**Comment by TheJaredWilcurt** [comment]
Well this hilarious and sad. I had to download Handbrake on a new machine, couldn't remember the last version that offered "Target Size", so I googled it. Someone on StackOverflow said "Just use VidCoder, it's the same thing, but with that extra feature".

Then I download VidCoder only to find out that I'm trading like 50 features for one. Genius.

Anyways, uninstalling VidCoder and going back to Handbrake 0.9.5.

Hilarious. But also sad. Stuck in 2011.

Before anyone says "just use the last **good** version of VidCoder instead of the last **good** version of Handbrake".

What's the point? Why leave one program where the author gave up on making a good app for another *based on it* where the author gave up too?

**Comment by RandomEngy** [comment]
1) The latest version of HandBrake doesn't have the Advanced tab either.
2) If you care to have a version with target size and Advanced tab, download [VidCoder 4.36](https://github.com/RandomEngy/VidCoder/releases/tag/v4.36) .
3) Everything in the Advanced tab is still possible by typing in the settings and referring to the [x264 documentation](https://en.wikibooks.org/wiki/MeGUI/x264_Settings). There are more settings available than you could cram into one window. Also it's never out of date.
4) It is very difficult to actually come up with settings that give you a better speed/compression efficiency tradeoff than the built-in presets/tunes. If you are at the technical level where you can improve on what the x264 developers have created, you very likely can deal with typing in the settings that you want.
